eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -223,14 +229,25 @@ object MultiPartHandler {
               val paymentPreimage = paymentPreimage_opt.getOrElse(randomBytes32())
               val paymentHash = Crypto.sha256(paymentPreimage)
               val expirySeconds = expirySeconds_opt.getOrElse(nodeParams.paymentRequestExpiry.toSeconds)
-              val features = {
-                val f1 = Seq(Features.PaymentSecret.mandatory, Features.VariableLengthOnion.mandatory)
-                val allowMultiPart = nodeParams.features.hasFeature(Features.BasicMultiPartPayment)
-                val f2 = if (allowMultiPart) Seq(Features.BasicMultiPartPayment.optional) else Nil
-                val f3 = if (nodeParams.enableTrampolinePayment) Seq(Features.TrampolinePayment.optional) else Nil
-                PaymentRequest.PaymentRequestFeatures(f1 ++ f2 ++ f3: _*)
+              val paymentMetadata = Some(ByteVector64.Zeroes.bytes)
+              val invoiceFeatures = {
+                val activatedInvoiceFeatures = nodeParams.features.invoiceFeatures().activated.map { case (f, s) => f.supportBit(s) }.toSet
+                val allInvoiceFeatures = if (nodeParams.enableTrampolinePayment) activatedInvoiceFeatures + Features.TrampolinePayment.optional else activatedInvoiceFeatures
+                allInvoiceFeatures.toSeq
```

Comment:
I'll try that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2063/commits/4c39fd36e1f78f7efdfbdcc533dab6069174043f
I typed `PaymentRequestFeatures.apply`, but I didn't use lenses.
Trampoline should be the only exception, it's mostly a hack because we don't want to advertise yet our trampoline support in `init` and `node_announcement` until the spec PR has more traction, but we'll remove that hack once trampoline is accepted.